index.js

```diff
@@ -1,55 +1,78 @@
-const {token} = require('./config.json');
-const Discord = require('discord.js');
-const client = new Discord.Client();
+const { token } = require('./config.json')
+const Discord = require('discord.js')
+const client = new Discord.Client()
 
 client.on('ready', () => {
-	console.log(`Logged in as ${client.user.tag}!`);
+    console.log(`Logged in as ${client.user.tag}!`)
 
-	//list servers bot is connected to
-	console.log("\nServers:");
-	client.guilds.cache.forEach((guild) => {
-		console.log(" - " + guild.name);
+    //list servers bot is connected to
+    console.log('\nServers:')
+    client.guilds.cache.forEach((guild) => {
+        console.log(' - ' + guild.name)
 
-		//list channels bot in on
-		guild.channels.cache.forEach((channel) => {
-			console.log(" -- " + channel.name + " (" + channel.type + ") - " + channel.id);
-		})
-	});
+        //list channels bot in on
+        guild.channels.cache.forEach((channel) => {
+            console.log(
+                ' -- ' +
+                    channel.name +
+                    ' (' +
+                    channel.type +
+                    ') - ' +
+                    channel.id
+            )
+        })
+    })
+})
 
- });
+client.on('message', (msg) => {
+    //console.log(msg);
+    var message = msg.content.toLowerCase()
+    var channelId = msg.channel.id
+    //console.log(msg.channel.id);
+    var channel = client.channels.cache.get(channelId)
+    if (message.startsWith('r!')) {
+        message = message.substring(2) //extracts the message without the "r!"
+        dicePattern = new RegExp('d+dd+/i')
+        if (message === 'beer') {
+            channel.send('yummy')
+        } else if (message === 'flops') {
+            channel.send('F')
+            channel.send(':clap::clap:    :clap::clap:')
+            channel.send('L')
+            channel.send(':clap::clap:    :clap::clap:')
+            channel.send('O')
+            channel.send(':clap::clap:    :clap::clap:')
+            channel.send('P')
+            channel.send(':clap::clap:    :clap::clap:')
+			channel.send('Flopsssssss')	
+        } else if (message.match(dicePattern)) { // checks if message is formatted like [numDice]d[numSides]
+            var xy = message.split('d')
+            var numDice = parseInt(xy[0])
+            var numSides = parseInt(xy[1])
 
-client.on('message', msg => {
-	//console.log(msg);
-	var message = msg.content.toLowerCase();
-	var channelId = msg.channel.id;
-	//console.log(msg.channel.id);
-	var channel = client.channels.cache.get(channelId);
-	if (message.startsWith('r!')){
-		 message = message.substring(2); //extracts the message without the "r!"
+            for (let i = 0; i < numDice; i++) {
+				channel.send('rolling dice ' + (i + 1))
+				// To add a bit of suspense lol
+                for (j = 0; j < 3; j++) {
+                    setTimeout(() => {
```

Comment:
I might be wrong, but I think that because setTimeout is what is called _asynchronous_, this might not act as you expect. The 3 setTimouts will trigger at almost the exact same time, because the program does not halt when starting a setTimeout. So the effect will be 3 dots coming at the same time in 500ms. Similarly, I think line 62 will fire almost immediately for similar reasons. 
If you are able to get it to wait correctly, it might just halt the entire bot, which is not something you might want. There is probably a way to make it work the way you want it to, but it's probably pretty complicated, since having tasks run in parallel gets pretty hard to track.